flow/envs/loop/lane_changing.py

```diff
@@ -210,14 +210,13 @@ def get_state(self):
                     lane_headways[j] /= max_length
                     vel_in_front[j] = self.vehicles.get_speed(lane_leader) \
                         / max_speed
+                    self.visible.extend([lane_leader])
             for j, lane_follower in enumerate(lane_followers):
                 if lane_follower != '':
                     lane_headways[j] /= max_length
                     vel_behind[j] = self.vehicles.get_speed(lane_follower) \
                         / max_speed
-
-            self.visible.extend(lane_leaders)
-            self.visible.extend(lane_followers)
+                    self.visible.extend([lane_follower])
```

Comment:
Are these changes just to more clearly show when the list of visible cars is being changed?

Reply:
this is to fix a bug where the list of visible vehicles sometimes includes '' strings